web/docs/advanced/links.md

```diff
@@ -109,16 +164,39 @@ export const routes = {
     }) => // ...
   },
 
-  // DetailRoute has a path like "/task/:id/:something?"
+  // DetailRoute has a path like "/task/:id/:userId?"
   DetailRoute: {
     build: (
       options: {
-        params: { id: ParamValue; something?: ParamValue; },
+        params: { id: ParamValue; userId?: ParamValue; },
         search?: string[][] | Record<string, string> | string | URLSearchParams
         hash?: string
       }
     ) => // ...
-  }
+  },
+
+  // OptionalRoute has a path like "/task/:id/details?"
+  OptionalRoute: {
+    build: (
+      options: {
+        path: '/task/:id/details' | '/task/:id',
```

Comment:
How come this is the only one that specifies `path`?

Reply:
For others, we know the exact string on which we will operate. When optional static segments are involved - we need the user to tell us which variant to use.